app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Admin::ApplicationsController < ApplicationController
+    def show
+        @application = Application.find(params[:id])
+        
+        @pets = Pet.search(params[:search])
+        
+        if params[:added_pet].present? 
+            pet_id = params[:added_pet]
+            @application.add_pet_to_application(pet_id)
+        end
+
+        if params[:pet_ownership_description].present?
+            @application.change_application_status("Pending")
+        end
+
+        if params[:approved_pet_id].present?
```

Comment:
This and the following if statement are adding a lot of functionality to this controller action. Great work figuring out how to add query parameters to these buttons! In the interest of SRP and REST, though, it would be good to break up this functionality a bit. 

For the "approve" and "reject" buttons, could you direct those to a PATCH route that updates the status of the pet_application? It could be the `update` action of the pet_applications_controller.